x-pack/plugins/reporting/export_types/csv_from_savedobject/server/lib/get_filters.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { badRequest } from 'boom';
+import moment from 'moment-timezone';
+
+import { SavedSearchObjectAttributes, TimeRangeParams } from '../../';
+import { QueryFilter, Filter, SearchSourceFilter } from './';
+
+export function getFilters(
+  indexPatternId: string,
+  indexPatternTimeField: string | null,
+  timerange: TimeRangeParams | null,
+  savedSearchObjectAttr: SavedSearchObjectAttributes,
+  searchSourceFilter?: SearchSourceFilter,
+  queryFilter?: QueryFilter
+) {
+  let includes: string[];
+  let timeFilter: any | null;
+  let timezone: string | null;
+
+  if (indexPatternTimeField) {
+    if (!timerange) {
+      throw badRequest(
+        `Time range params are required for index pattern [${indexPatternId}], using time field [${indexPatternTimeField}]`
+      );
+    }
+
+    timezone = timerange.timezone;
+    const { min: gte, max: lte } = timerange;
+    timeFilter = {
+      range: {
+        [indexPatternTimeField]: {
+          format: 'strict_date_time',
+          gte: moment.tz(moment(gte), timezone).format(),
+          lte: moment.tz(moment(lte), timezone).format(),
```

Comment:
This is overcomplicated. The `timezone` value here is meant to be used to format date values in their CSV download.

I don't think Elasticsearch needs or wants the date formatted for random timezone.